app/src/policy_engine.h

```diff
@@ -0,0 +1,237 @@
+
+      // Vanilla SCITT protected header parameters
+      {
+        if (phdr.alg.has_value())
+        {
+          obj.set_int64("alg", phdr.alg.value());
+        }
+
+        if (phdr.crit.has_value())
+        {
+          auto crit_array = ctx.new_array();
+          size_t i = 0;
+
+          for (const auto& e : phdr.crit.value())
+          {
+            if (std::holds_alternative<int64_t>(e))
+            {
+              crit_array.set_at_index(
+                i++,
+                ccf::js::core::JSWrappedValue(
+                  ctx, JS_NewInt64(ctx, std::get<int64_t>(e))));
+            }
+            else if (std::holds_alternative<std::string>(e))
+            {
+              crit_array.set_at_index(
+                i++, ctx.new_string(std::get<std::string>(e)));
+            }
+          }
+
+          obj.set("crit", std::move(crit_array));
+        }
+
+        if (phdr.kid.has_value())
+        {
+          obj.set("kid", ctx.new_string(phdr.kid.value()));
+        }
+
+        if (phdr.issuer.has_value())
+        {
+          obj.set("issuer", ctx.new_string(phdr.issuer.value()));
+        }
+
+        if (phdr.feed.has_value())
+        {
+          obj.set("feed", ctx.new_string(phdr.feed.value()));
+        }
+
+        if (phdr.cty.has_value())
+        {
+          if (std::holds_alternative<int64_t>(phdr.cty.value()))
+          {
+            obj.set_int64("cty", std::get<int64_t>(phdr.cty.value()));
+          }
+          else if (std::holds_alternative<std::string>(phdr.cty.value()))
+          {
+            obj.set(
+              "cty", ctx.new_string(std::get<std::string>(phdr.cty.value())));
+          }
+        }
+
+        if (phdr.x5chain.has_value())
+        {
+          auto x5_array = ctx.new_array();
+          size_t i = 0;
+
+          for (const auto& der_cert : phdr.x5chain.value())
+          {
+            auto pem = ccf::crypto::cert_der_to_pem(der_cert);
+            x5_array.set_at_index(i++, ctx.new_string(pem.str()));
+          }
+
+          obj.set("x5chain", std::move(x5_array));
+        }
+      }
+
+      // Extra Notary protected header parameters.
+      {
+        if (phdr.notary_signing_scheme.has_value())
+        {
+          obj.set(
+            "notary_signing_scheme",
+            ctx.new_string(phdr.notary_signing_scheme.value()));
+        }
+
+        if (phdr.notary_signing_time.has_value())
+        {
+          obj.set_int64(
+            "notary_signing_time", phdr.notary_signing_time.value());
+        }
+
+        if (phdr.notary_authentic_signing_time.has_value())
+        {
+          obj.set_int64(
+            "notary_authentic_signing_time",
+            phdr.notary_authentic_signing_time.value());
+        }
+
+        if (phdr.notary_expiry.has_value())
+        {
+          obj.set_int64("notary_expiry", phdr.notary_expiry.value());
+        }
+      }
+
+      return obj;
+    }
+
+    static inline std::optional<std::string> apply_js_policy(
```

Comment:
Please add unit tests to make sure we have coverage of the edge cases, see examples: 
- https://github.com/microsoft/scitt-ccf-ledger/blob/main/app/unit-tests/configurable_auth_test.cpp
- https://github.com/microsoft/scitt-ccf-ledger/blob/main/app/unit-tests/verifier_test.cpp

Reply:
As discussed, added tests for:

1. Invalid JS in body of policy
2. Timeout/infinite loop
3. Heap and Stack limits